scripts/make.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+import platform
+import subprocess
+import os
+
+class build:
+    def __init__(self, cmd_args):
+        self._build_dir = 'build'
+        self._generator = 'Ninja'
+        self._extra_gen_args = []
+        self._extra_build_args = []
+        self._verbose = cmd_args.verbose
+        self._tests_target = 'test'
+
+    def generate(self):
+        args = ['cmake', '-S', '.', '-B', self._build_dir, '-G', self._generator]
+        args.extend(self._extra_gen_args)
+        self._execute(args)
+
+    def make(self):
+        args = ['cmake', '--build', self._build_dir]
+        args.extend(self._extra_build_args)
+        if self._verbose:
+            args.append('-v')
+        
+        self._execute(args)
+
+    def test(self):
+        args = ['cmake', '--build', self._build_dir]
+        args.extend(self._extra_build_args)
+        args.extend(['--target', self._tests_target])
+        
+        self._execute(args)
+
+    def clean(self):
+        args = ['cmake', '--build', self._build_dir]
+        args.extend(self._extra_build_args)
+        args.extend(['--target', 'clean'])
+
+        self._execute(args)
+        
+    def run_all(self):
+        self.generate()
+        self.make()
+        self.test()
+
+    def _execute(self, args):
+        subprocess.check_call(args)
+
+class windows_build(build):
```

Comment:
add `single_config_build` and `multi_config_build` which will properly set the additional build arguments; derive from those classes, setting gen args and tests target